pkg/types/currency_pair.go

```diff
@@ -6,17 +6,24 @@ import (
 )
 
 const (
-	ethereum = "ETHEREUM"
+	ethereum         = "ETHEREUM"
+	DefaultDelimiter = "/"
 )
 
 // NewCurrencyPair returns a new CurrencyPair with the given base and quote strings.
 func NewCurrencyPair(base, quote string) CurrencyPair {
 	return CurrencyPair{
-		Base:  base,
-		Quote: quote,
+		Base:      base,
+		Quote:     quote,
+		Delimiter: DefaultDelimiter,
 	}
 }
 
+func (cp CurrencyPair) WithDelimiter(delimiter string) CurrencyPair {
+	cp.Delimiter = delimiter
+	return cp
+}
+
 // ValidateBasic checks that the Base / Quote strings in the CurrencyPair are formatted correctly, i.e.
 // Base + Quote are non-empty, and are in upper-case.
 func (cp *CurrencyPair) ValidateBasic() error {
```

Comment:
Is there a set of valid characters we should allow as delimiters?

Reply:
we should also limit it to be a string of `len == 1`

Reply:
Yeah right now I'm just considering the canonical implementation as `/` but obviously some people use `-` or I guess you could use `_` or conceivable anything else that's not also expected to be included in a base or quote.